File: LeetCode/1. Two_Sum.rb
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

def two_sum(nums, target)
  nums.each_with_index do |el1, idx1|
     nums.each_with_index do |el2, idx2|
        if idx2 > idx1 && el1 + el2 == target
            return [idx1, idx2]
        end
     end
  end
end

puts two_sum([2,7,11,15], 9)

# Output
# [0,1]
# Expected
# [0,1]

File: LeetCode/js/1_Two_Sum.js
// Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

// You may assume that each input would have exactly one solution, and you may not use the same element twice.

// You can return the answer in any order.

 

// Example 1:

// Input: nums = [2,7,11,15], target = 9
// Output: [0,1]
// Output: Because nums[0] + nums[1] == 9, we return [0, 1].
// Example 2:

// Input: nums = [3,2,4], target = 6
// Output: [1,2]
// Example 3:

// Input: nums = [3,3], target = 6
// Output: [0,1]

var twoSum = function(nums, target) {
  let result;
  for(let i = 0; i < nums.length-1; i++){
      for(let j = i+1; j < nums.length; j++){
          if(nums[i] + nums[j] == target){
             result = [i,j]
          }
      }
  }
     return result
  };

  console.log(twoSum([2,7,11,15], 9))